### PR TITLE
feat: add versioning for genesis generation

### PR DESCRIPTION
**Pull request by lubtd**
Close https://github.com/ignite/cli/issues/2260

- Add field `cli.version` in `metadata`
- Set version to `1`
- Verify version for commands generating the genesis: `chain show genesis` and `chain prepare`
- Set the version automatically in metadata on `publish`

If the version is different
- `chain show genesis` just show a warning since it is the advanced workflow
- `prepare` fails unless `--force` is used

### CODE REVIEW

File: network/publish_test.go [hunk 1]
```diff
@@ -359,6 +365,7 @@ func TestPublish(t *testing.T) {
 					),
 					HasCampaign: false,
 					CampaignID:  0,
+					Metadata:    []byte("{\"cli\":{\"version\":\"1\"}}"),
```
Comment (jeronimoalbi) [inline network/publish_test.go:368]: Shouldn't it use the `metadata` variable?

```suggestion
					Metadata:    metadata,
```

File: network/publish_test.go [hunk 2]
```diff
@@ -473,6 +481,7 @@ func TestPublish(t *testing.T) {
...
```
Comment (jeronimoalbi) [inline network/publish_test.go:484]: Same as above
```suggestion
					Metadata:    metadata,
```

File: network/publish_test.go [hunk 3]
```diff
@@ -20,6 +20,8 @@ import (
 	"github.com/ignite/cli-plugin-network/network/testutil"
 )
 
+var metadata = []byte("{\"cli\":{\"version\":\"1\"}}")
```
Comment (tbruyelle) [inline network/publish_test.go:23]:
```suggestion
var metadata = []byte(`{"cli":{"version":"1"}}`)
```

Easier to read no?

File: cmd/network_chain_show_genesis.go [hunk 4]
```diff
@@ -95,5 +96,15 @@ func networkChainShowGenesisHandler(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	if chainLaunch.Metadata.Cli.Version != "" && chainLaunch.Metadata.Cli.Version != networktypes.Version {
```
[the remark this answers is not in the document]
Comment (lubtd) [reply cmd/network_chain_show_genesis.go]: [bd71858](https://github.com/ignite/cli-plugin-network/pull/9/commits/bd71858fe454c7bab694c212d0271d61ad64bb8c)

**Comment by lubtd** [comment]
@tbruyelle ready for review

**Review by tbruyelle — APPROVED** [review]

**Review by jeronimoalbi — APPROVED** [review]

**Pull request merged**